pkg/scheduler/eventhandlers.go

```diff
@@ -492,37 +492,49 @@ func nodeSpecUnschedulableChanged(newNode *v1.Node, oldNode *v1.Node) bool {
 }
 
 func preCheckForNode(nodeInfo *framework.NodeInfo) queue.PreEnqueueCheck {
-	// In addition to the checks in kubelet (pkg/kubelet/lifecycle/predicate.go#GeneralPredicates),
-	// the following logic appends a taint/toleration check.
-	// TODO: verify if kubelet should also apply the taint/toleration check, and then unify the
-	// logic with kubelet and move to a shared place.
-	//
 	// Note: the following checks doesn't take preemption into considerations, in very rare
 	// cases (e.g., node resizing), "pod" may still fail a check but preemption helps. We deliberately
 	// chose to ignore those cases as unschedulable pods will be re-queued eventually.
 	return func(pod *v1.Pod) bool {
-		if len(noderesources.Fits(pod, nodeInfo, feature.DefaultFeatureGate.Enabled(features.PodOverhead))) != 0 {
+		_, isUntolerated := corev1helpers.FindMatchingUntoleratedTaint(nodeInfo.Node().Spec.Taints, pod.Spec.Tolerations, func(t *v1.Taint) bool {
+			return t.Effect == v1.TaintEffectNoExecute
```

Comment:
> You can refer to #104421 to (1) exclude taint/toleration check in GeneralFilter(), and (2) run taint/toleration check in scheduler#preCheckForNode() specifically - note that it shoule only check **NoSchedule** effect.

The scheduler should consider the `NoSchedule` effect.
`NoExecute` effect will be considered at the kubelet.

And I think we need to update the tests.

Reply:
Yes, `NoSchedule` for scheduler side. And some UTs need to be updated.

Reply:
Sorry this is a hand error : )